### DISCUSSION TITLE
System dark/light mode

### DISCUSSION THREAD

**Discussion by inari-codes** [beercss/beercss#309] [General]
Hello, came back from a long time :D 
Currently I’m making an simple note taking app, and I want it to respond to dark mode/light mode changes but by default BeerCSS doesn’t do it automatically and (maybe) no other way except making an style.css file and add the color values in there which is quite cumbersome. So is there an easier way to do it?

**Comment by KicKerBNU** [comment]
Hello @ItsKruistz  

Actually we have a dark and light in beercss you can test it here:

On the top right of the website you have a moon icon to click and turn it to dark / light
https://www.beercss.com/

We have also an codepen example here by clicking in the `light/dark` button:
https://codepen.io/kickerbnu/pen/Rwogbpe

Hope it helps and have a good sunday 🍻

**Reply by inari-codes** [reply]
Yes, I know there's a dark theme 😉 But what I meant is making it change to dark or light theme based on system setting

**Reply by KicKerBNU** [reply]
Well, you can look at the user browser to see if its dark

const isSystemDark = window.matchMedia("(prefers-color-scheme: dark)").matches
console.log(isSystemDark);

Based on this information you can set dark or light in your app

**Reply by samueller** [reply]
Is there a way to automatically switch to dark/light mode based on the OS setting? For example, my macOS switches my windows (and some websites) to dark mode during sunset. Is there an event I can listen to in order to trigger the mode switch?

**Reply by leonardorafael** [reply]
I would try to get the sunrise and sunset times of your location (on page load) and check it every minute:

```javascript
setInterval(() => {
  if (currentTime > sunriseTime && currentTime < sunsetTime) ui("mode", "light");
  else ui("mode", "dark");
}, 60000);
```

But I don't know if this information already exists on browser.

**Reply by leonardorafael** [reply]
Maybe if you put the code from @KicKerBNU inside interval, could work

```javascript
setInterval(() => {
  const isSystemDark = window.matchMedia("(prefers-color-scheme: dark)").matches;
  if (isSystemDark) ui("mode", "dark");
  else ui("mode", "light");
}, 60000);
```

**Reply by samueller** [reply]
I just learned about [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme). I'm surprised Beer CSS doesn't use this in CSS in order to switch light-dark modes automatically. I got everything to work with the following CSS:
```css
:root {
	--primary: #695f00;
	--on-primary: #ffffff;
	--primary-container: #f9e534;
	/* ... */
}
@media (prefers-color-scheme: dark) {
	:root {
		--primary: #dbc90a;
		--on-primary: #363100;
		--primary-container: #4f4800;
		/* ... */
	}
}
```

**Reply by leonardorafael** [reply]
Sorry, I understand wrong your question. I understood that you would change the mode without refresh page. 😅
I think we can do that, but I don't know if this needs to be the default.

**Reply by samueller** [reply]
To me, this seems like the expected default behavior. I was actually surprised when the light/dark switch didn't happen automatically. And then confused why I didn't see anything about it in the docs. But this is just my opinion...
